fix: stuck buildVersion spinner

Mika412 opened the pull request:
The main issue with the spinner, is that it read from the pipe, before the process finished writing everything. In some cases this resulted in incomplete data. 

I've also refactored some things, disabled the items while there's already a running modification, and reset the build number if an invalid input is provided. 

Another alternative, to speed up the value loading, would be to save them in the Bottle class.


https://github.com/IsaacMarovitz/Whisky/assets/5073663/2df8b42b-1073-4432-ac94-39a82ba461c4


When the values fail to load:

![image](https://github.com/IsaacMarovitz/Whisky/assets/5073663/3ba6052f-9d44-40de-83ef-f14f09bbe84f)


Partially fixes #159

Mika412 commented:
#157

IsaacMarovitz commented:
For me this results in the Retina Mode toggle being perpetually disabled, and I noticed a "Wine crashed! (1)" log

Mika412 commented:
Figured out the problem. It seems that the process "crashes" when it can't find the registry key, with a message "Unable to find the specified registry key" in the output, that was ignored. Guess the retina key was only created when you toggle it, and not during the bottle creation. 

I've added a retina key check. If it doesn't exist, it creates it. I've also added non existing key handling, so instead of throwing an error it returns an optional default value.
